openedxscorm/scormxblock.py

```diff
@@ -326,20 +358,33 @@ def extract_package(self, package_file):
                             dest_path,
                             ContentFile(scorm_zipfile.read(zipinfo.filename)),
                         )
-
+        
     @property
     def index_page_url(self):
         if not self.package_meta or not self.index_page_path:
             return ""
-        folder = self.extract_folder_path
-        if self.storage.exists(
-            os.path.join(self.extract_folder_base_path, self.index_page_path)
-        ):
-            # For backward-compatibility, we must handle the case when the xblock data
-            # is stored in the base folder.
-            folder = self.extract_folder_base_path
-            logger.warning("Serving SCORM content from old-style path: %s", folder)
-        return self.storage.url(os.path.join(folder, self.index_page_path))
+
+        querystring_auth_setting = self.xblock_settings.get('SCORM_S3_QUERY_AUTH')
+        if querystring_auth_setting is False:
+            folder = self.extract_folder_path
+            if self.storage.exists(
+                os.path.join(self.extract_folder_base_path, self.index_page_path)
+            ):
+                # For backward-compatibility, we must handle the case when the xblock data
+                # is stored in the base folder.
+                folder = self.extract_folder_base_path
+                logger.warning("Serving SCORM content from old-style path: %s", folder)
+            return self.storage.url(os.path.join(folder, self.index_page_path))
+        else:
+            url = self.runtime.handler_url(self, 'scorm_view')
+            if url.endswith('?'):
```

Comment:
I don't think `handler_url` returns a url with `?` in it. Could you please describe which scenario has `?`?